leader/ring/korach_rotem_santoro_stagesWithFeedback.go

```diff
@@ -0,0 +1,277 @@
+package ring
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"github.com/krzysztof-turowski/distributed-framework/lib"
+	"log"
+)
+
+const (
+	msgE = iota
+	msgA
+	msgT
+)
+const (
+	statusECandidate = iota
+	statusACandidate
+	statusPassive
+	statusElected
+)
+const (
+	left = iota
+	right
+	both
+)
+const none = -1
+const byteNone = 255
+
+/*
+	in our model nodes do not run continuously,
+	as such status "Available" specified in original algorithm does not exist,
+	all nodes initiate the election and so all enter as E-Candidate
+*/
+
+type Message struct {
+	Type byte
+	Val  int64
+}
+
+type stateStagesWithFeedback struct {
+	Status             int
+	Leader             int64
+	Id                 int64
+	Messages           []Message
+	ReverseOrientation bool
```

Comment:
Messages are needed, because active node needs messages from both sides at once to proceed. Leader and Orientation are used in checkSingleLeaderElected to check if algorithm worked. Id was recently removed.